cabal-install/Main.hs

```diff
@@ -258,6 +256,7 @@ mainWorker args = topHandler $
       , regularCmd runCommand runAction
       , regularCmd initCommand initAction
       , regularCmd configureExCommand configureAction
+      , regularCmd reconfigureCommand reconfigureAction
```

Comment:
Based on conversation with @ttuegel, I think one reason you would call `reconfigure` directly is to modify the set of configuration parameters slightly, while still preserving the old set.